src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationResolverOptionsSetup.cs

```diff
@@ -62,41 +93,195 @@ private string GetWorkersRootDirPath(IConfiguration configuration)
+        /// Logs a trace message if the required configuration section is found.
+        /// </summary>
+        private void LogWorkersDirSectionPresence(IConfiguration configuration, string configurationSource)
+        {
+            string configSectionToCheck = $"{RpcWorkerConstants.LanguageWorkersSectionName}:{WorkerConstants.WorkersDirectorySectionName}";
+            var section = configuration.GetSection(configSectionToCheck);
+
+            if (!string.IsNullOrEmpty(section.Value))
+            {
+                _logger.LogTrace("Found configuration section '{requiredSection}' in '{configurationSource}'.", configSectionToCheck, configurationSource);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves the list of worker probing paths from configuration or uses the default path for Windows environment.
+        /// </summary>
+        internal List<string> GetWorkerProbingPaths(IConfiguration configuration)
+        {
+            // If Configuration section is set, read probing paths from configuration.
+            var probingPathsSection = configuration.GetSection($"{RpcWorkerConstants.LanguageWorkersSectionName}:{RpcWorkerConstants.WorkerProbingPathsSectionName}");
+            var subSectionsCount = probingPathsSection.GetChildren().Count();
+            var probingPaths = new List<string>();
+
+            if (subSectionsCount == 0)
+            {
+                if (_environment.IsHostedWindowsEnvironment())
+                {
+                    // Default worker probing path for Windows
+                    string windowsSiteExtensionsPath = GetWindowsSiteExtensionsPath();
+
+                    if (!string.IsNullOrWhiteSpace(windowsSiteExtensionsPath))
+                    {
+                        // Example probing path for Windows: "c:\\home\\SiteExtensions\\functionsworkers"
+                        var windowsWorkerProbingPath = Path.Combine(windowsSiteExtensionsPath, RpcWorkerConstants.FunctionsWorkersDirectoryName);
+                        probingPaths.Add(windowsWorkerProbingPath);
+                    }
+                }
+
+                return probingPaths;
+            }
+
+            for (int i = 0; i < subSectionsCount; i++)
+            {
+                var path = probingPathsSection.GetSection($"{i}").Value;
+                if (!string.IsNullOrWhiteSpace(path))
+                {
+                    probingPaths.Add(path);
+                }
+            }
+
+            return probingPaths;
+        }
+
+        /// <summary>
+        /// Returns the default site extensions path for Windows environment.
+        /// </summary>
+        private string GetWindowsSiteExtensionsPath()
+        {
+            var assemblyDir = AppContext.BaseDirectory;
+
+            //Move 2 directories up to get to the SiteExtensions directory. Example output: "c:\\home\\SiteExtensions"
+            return Directory.GetParent(assemblyDir.TrimEnd(Path.DirectorySeparatorChar))?.Parent?.FullName;
```

Comment:
Can we validate here if we indeed are reaching the Site Extensions Directory?

Reply:
I am logging probing paths in `DynamicWorkerConfigurationResolver`. That will help us identify in case of issues. Also, if the probing path is not correct or does not exist then code will gracefully fallback to load worker configs from within the Host.